internal/pkg/configuration/configuration.go

```diff
@@ -86,7 +87,20 @@ func Generate(ctx context.Context, in *machine.GenerateConfigurationRequest) (re
 			return nil, err
 		}
 
-		reply.Data = [][]byte{cfgBytes}
+		talosconfig, err := generate.Talosconfig(input, options...)
+		if err != nil {
+			return nil, err
+		}
+
+		taloscfgBytes, err = talosconfig.Bytes()
+
+		if err != nil {
+			return nil, err
+		}
+
+		reply = &machine.GenerateConfigurationResponse{
+			Data: [][]byte{cfgBytes, taloscfgBytes},
```

Comment:
It feels fragile to have the configs in a slice. Ordering becomes important at that point.

Reply:
probably deserves a field on its own

Reply:
:heavy_check_mark: